### ISSUE TITLE
Syntax Coloring Does Not Color Javascript Within HTML FIle Correctly

### ISSUE THREAD

**Issue by tbpassin** [leo-editor/leo-editor#3281]
See PR #3282.

Script elements within an HTML file don't get colorized as javascript.  Instead, either they don't get colorized  or the HTML colorizer gets very confused. 

**Background (EKR)**

The `jedit2py` script in `scripts.leo` transliterated all of jEdit's XML files to python. The intention of all the python files in `leo/modes` is to simulate jEdit's colorizing rules. The python files do this in close cooperation with Leo's jEdit colorizer in `leoColorizer.py`.

`leo/modes/html.py` delegates coloring for `<script>` elements to `leo/modes/javascript.py` with the following rule:
```python
def html_rule1(colorer, s, i):
    return colorer.match_span(s, i, kind="markup", begin="<SCRIPT", end="</SCRIPT>",
        at_line_start=False, at_whitespace_end=False, at_word_start=False,
        delegate="html::javascript", exclude_match=False,
        no_escape=False, no_line_break=False, no_word_break=False)
```
Most of this rule is boilerplate. All the kwargs look good. The only important line is:
```python
    return colorer.match_span(s, i, kind="markup", begin="<SCRIPT", end="</SCRIPT>",
```
The following bugs are apparent in the above line:
- `jedit.match_span` does an exact match, so it will match neither `<script>` nor `</script>`.
- Attributes of the `script` element will be colorized as JS, not html.
- The closing `>` of the '<script>` element will be colorized as JS, not html.

Bugs can arise in various ways:
1. In the rule itself. See above.
2. Leo might not handle delegation properly.
3. In the rules in the delegated colorer, `javascript.py` in this example.

**Debugging**: The `--trace=coloring` command-line argument provides a superb trace.

**Comment by edreamleo** [comment]
@tbpassin Please send me an example that illustrates the problem. Thanks.

**Comment by tbpassin** [comment]
Here is the example in which I first noticed the issue.  Scroll down to around line 70.  Before that, none of the js code has been colorized.  From that point on, the colorizer is very confused.

[js-in-html-colorizer-example.zip](https://github.com/leo-editor/leo-editor/files/11261164/js-in-html-colorizer-example.zip)

**Comment by edreamleo** [comment]
@tbpassin Thanks. I have the file. This issue is much more complicated than I thought. I'll get to it when I can.

**Comment by edreamleo** [comment]
Closed via PR #3282.
